inc/notificationtargetchange.class.php

```diff
@@ -203,15 +201,16 @@ function getDataForObject(CommonDBTM $item, array $options, $simple = false) {
          $data['##change.numberofitems##'] = count($data['items']);
 
          //Validation infos
-         $restrict = "`changes_id`='".$item->getField('id')."'";
-
          if (isset($options['validation_id']) && $options['validation_id']) {
-            $restrict .= " AND `glpi_changevalidations`.`id` = '".$options['validation_id']."'";
+            $restrict['glpi_changevalidations.id'] = $options['validation_id'];
          }
 
-         $restrict .= " ORDER BY `submission_date` DESC, `id` ASC";
-
-         $validations = getAllDatasFromTable('glpi_changevalidations', $restrict);
+         $validations = getAllDatasFromTable(
+            'glpi_changevalidations',
+            $restrict,
+            true,
```

Comment:
`$useCache` was using `false` default value.

Reply:
If conditiion is not empty, result will be the same; cache will not be used. If there is no condition (not sure this can heppens here), we can use the cache in current case.